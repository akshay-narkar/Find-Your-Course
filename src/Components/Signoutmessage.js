import { Redirect, useHistory } from 'react-router-dom';
import { signout } from '../API/apicalls';
import Navbar from './Navbar';
import helperauth from '../API/helper_auth';

export default function Signout() {
  const history = useHistory();

  const signout1 = () => {
    // const sessh = JSON.parse(sessionStorage.getItem('uid'));

    // const params = {
    //   uid: sessh.uid,
    //   client: sessh.client,
    //   'access-token': sessh['access-token'],
    // };
    let params;
    if (sessionStorage.getItem('uid')) { params = helperauth(); }

    signout(params, history);
  };

  return (
    <>
      {!sessionStorage.getItem('uid') ? (
        <>
          <Redirect to="/login" />
          {console.log('take me home')}
        </>
      )
        : (
          <>
            <Navbar />
            {signout1()}

          </>
        )}
    </>
  );
}
