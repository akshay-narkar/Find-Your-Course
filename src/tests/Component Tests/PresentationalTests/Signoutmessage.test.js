import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Signoutmessage from '../../../Components/Signoutmessage';

it('UI is unaltered wrt first snapshot', () => {
  const elem = renderer.create(<BrowserRouter><Signoutmessage /></BrowserRouter>).toJSON();

  expect(elem).toMatchSnapshot();
});
