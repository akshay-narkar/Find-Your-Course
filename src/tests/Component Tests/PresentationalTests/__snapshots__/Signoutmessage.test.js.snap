// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI is unaltered wrt first snapshot 1`] = `null`;
